src/DssDeploy.t.sol

```diff
@@ -904,6 +924,54 @@ contract DssDeployTest is DssDeployTestBase {
         tusdJoin.exit(address(this), 10 ether);
     }
 
+    function testFailGemJoin7JoinWad() public {
+        DSValue pip = new DSValue();
+        USDT usdt = new USDT(100 * 10 ** 6);
+        GemJoin7 usdtJoin = new GemJoin7(address(vat), "USDT", address(usdt));
+        dssDeploy.deployCollateral("USDT", address(usdtJoin), address(pip));
+        usdt.approve(address(usdtJoin), uint(-1));
+        // Fail here
+        usdtJoin.join(address(this), 10 ether);
+    }
+
+    function testFailGemJoin7ExitWad() public {
+        DSValue pip = new DSValue();
+        USDT usdt = new USDT(100 * 10 ** 6);
+        GemJoin7 usdtJoin = new GemJoin7(address(vat), "USDT", address(usdt));
+        dssDeploy.deployCollateral("USDT", address(usdtJoin), address(pip));
+        usdt.approve(address(usdtJoin), uint(-1));
+        usdtJoin.join(address(this), 10 * 10 ** 6);
+        // Fail here
+        usdtJoin.exit(address(this), 10 ether);
+    }
+
+    function testFailGemJoin7Join() public {
+        DSValue pip = new DSValue();
+        USDT usdt = new USDT(100 * 10 ** 6);
+        GemJoin7 usdtJoin = new GemJoin7(address(vat), "USDT", address(usdt));
+        dssDeploy.deployCollateral("USDT", address(usdtJoin), address(pip));
+        usdt.approve(address(usdtJoin), uint(-1));
+        assertEq(usdt.balanceOf(address(this)), 100 * 10 ** 6);
+        assertEq(usdt.balanceOf(address(usdtJoin)), 0);
+        assertEq(vat.gem("USDT", address(this)), 0);
+        usdt.deprecate(address(0));
+        // Fail here
+        usdtJoin.join(address(this), 10 * 10 ** 6);
+    }
+
+    function testFailGemJoin7Exit() public {
+        DSValue pip = new DSValue();
+        USDT usdt = new USDT(100 ether);
+        GemJoin7 usdtJoin = new GemJoin7(address(vat), "USDT", address(usdt));
+        dssDeploy.deployCollateral("USDT", address(usdtJoin), address(pip));
+        usdt.approve(address(usdtJoin), uint(-1));
+        usdtJoin.join(address(this), 10 * 10 ** 6);
+        usdt.deprecate(address(0));
```

Comment:
I would choose a different address. The `upgradedAddress` is `0x000...` by default, so setting it to zero address here doesn't change it.

Reply:
resolved https://github.com/makerdao/dss-deploy/pull/53/commits/53c06b3097d6f8513b6dd7447df44b029fbb3105 

address(1)